packages/twenty-front/src/modules/ui/layout/modal/components/EnhancedModalLayout.tsx

```diff
@@ -0,0 +1,252 @@
+  <StyledHeader className={className}>{children}</StyledHeader>
+);
+
+type ModalLayoutContentProps = React.PropsWithChildren & {
+  className?: string;
+};
+
+const ModalLayoutContent = ({
+  children,
+  className,
+}: ModalLayoutContentProps) => (
+  <StyledContent className={className}>{children}</StyledContent>
+);
+
+type ModalLayoutFooterProps = React.PropsWithChildren & {
+  className?: string;
+};
+
+const ModalLayoutFooter = ({ children, className }: ModalLayoutFooterProps) => (
+  <StyledFooter className={className}>{children}</StyledFooter>
+);
+
+/**
+ * Modal
+ */
+export type ModalSize = 'small' | 'medium' | 'large';
+export type ModalPadding = 'none' | 'small' | 'medium' | 'large';
+export type ModalVariants = 'primary' | 'secondary';
+
+export type EnhancedModalLayoutProps = React.PropsWithChildren & {
+  size?: ModalSize;
+  padding?: ModalPadding;
+  className?: string;
+  hotkeyScope?: ModalHotkeyScope;
+  onEnter?: () => void;
+  modalVariant?: ModalVariants;
+} & (
+    | { isClosable: true; onClose: () => void }
+    | { isClosable?: false; onClose?: never } // or the absence of `isClosable`
+  );
+
+const modalAnimation = {
+  hidden: { opacity: 0 },
+  visible: { opacity: 1 },
+  exit: { opacity: 0 },
+};
+
+// This component should be used over Modal when seeking a modal feel without modal state (hotkeyScope etc)
+export const EnhancedModalLayout = ({
+  children,
+  size = 'medium',
+  padding = 'medium',
+  className,
+  hotkeyScope = ModalHotkeyScope.Default,
+  onEnter,
+  isClosable = false,
+  onClose,
+  modalVariant = 'primary',
+}: EnhancedModalLayoutProps) => {
+  const isMobile = useIsMobile();
+  const modalRef = useRef<HTMLDivElement>(null);
+
+  const {
+    goBackToPreviousHotkeyScope,
+    setHotkeyScopeAndMemorizePreviousScope,
+  } = usePreviousHotkeyScope();
+
+  useEffect(() => {
+    setHotkeyScopeAndMemorizePreviousScope(hotkeyScope);
+    return () => {
+      goBackToPreviousHotkeyScope();
+    };
+  }, [
+    hotkeyScope,
+    setHotkeyScopeAndMemorizePreviousScope,
+    goBackToPreviousHotkeyScope,
+  ]);
+
+  useScopedHotkeys(
+    [Key.Enter],
+    () => {
+      onEnter?.();
+    },
+    hotkeyScope,
+  );
+
+  useScopedHotkeys(
+    [Key.Escape],
+    () => {
+      if (isClosable && onClose !== undefined) {
+        onClose();
+      }
+    },
+    hotkeyScope,
+  );
+
+  useListenClickOutsideV2({
+    refs: [modalRef],
+    listenerId: 'MODAL_CLICK_OUTSIDE_LISTENER_ID',
+    callback: () => {
+      if (isClosable && onClose !== undefined) {
+        onClose();
+      }
+    },
+  });
+
+  const stopEventPropagation = (e: React.MouseEvent) => {
+    e.stopPropagation();
+  };
+
+  return (
+    <StyledBackDrop
+      onMouseDown={stopEventPropagation}
+      modalVariant={modalVariant}
+    >
+      <StyledModalDiv
+        // framer-motion seems to have typing problems with refs
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
```

Comment:
We try to avoid @ts-ignore, and in that case I don't have any error on my side?
<img width="171" alt="Screenshot 2024-08-07 at 09 42 44" src="https://github.com/user-attachments/assets/caf9b8af-1428-442a-81c3-6c6eb561e28d">

If you use VSCode you can try cmd+shift+p -> "Reload window", sometimes it helps when Typescript says something wrong

Reply:
okay, Got it!